python/scripts/rfi_receiver/rfi_receiver.py

```diff
@@ -822,18 +855,21 @@ def rfi_zeroing():
         time.sleep(1)
     while True:
         # Wait until the correct UTC time of the solar transit at DRAO (deals with daylight savings time)
-        t_now = datetime.datetime.utcnow()
-        t_transit = ephemeris.solar_transit(t_now)[0]
-        t_diff = datetime.datetime.utcfromtimestamp(t_transit) - t_now
+        today = datetime.datetime.utcnow().date()
+        t_day_start = datetime.datetime(today.year, today.month, today.day)
+        t_transit = ephemeris.solar_transit(t_day_start)[0]
+        t_diff = (
+            datetime.datetime.utcfromtimestamp(t_transit) - datetime.datetime.utcnow()
+        )
 
         time_to_transit_s = abs(t_diff.total_seconds())
 
         # Check if we are in the transit window, if so set downtime accordingly
         if time_to_transit_s < half_downtime_s:
+            toggle_rfi_zeroing(url, headers, False)
             downtime_s = time_to_transit_s + half_downtime_s
```

Comment:
This is wrong I think.

Consider what happens if you enter this before the transit window has ended, but after the transit itself.

In that case t_diff is negative, and you should only sleep for what's remaining in the window (i.e. `half_downtime_s + t_diff.total_seconds()`). But because of the abs in the definition of time_to_transit_s it will sleep too long.
That kind of mistake could very much give the asymmetry in the window.